### DISCUSSION TITLE
Cellular Networks configuration (LTE, LTE+, 5G-NR) of networks.json and regions.json files with Mosaic Cell in Eclipse MOSAIC

### DISCUSSION THREAD

**Discussion by Jean-Tshibangu-jtm** [eclipse/mosaic#371] [Q&A]
Hello,

I'm going to configure my cellular network using the networks.json and regions.json files. According to the documentation, the Mosaic Cell simulator is integrated into Eclipse MOSAIC.

Question 1: Is Mosaic Cell based on the cellular network (LTE, LTE +, ....)?

Question 2: Is it possible to configure LTE(LTE+) or 5G-NR from networks.json files?

Question 3: Is it possible to configure parameters such as Modulation, Receiver sensitivity power, Transmission Time Interval in the regions.json or networks.json files?

Question 4: Is it possible to configure the cell(regions.json) with 5G NR or LTE+ characteristics?

**Comment by schwepmo** [comment]
The Cell simulator has been build with runtime in mind and that's why it uses a *coarse* model for the cellular network.
So what you can do is define regions that can act as cells but don't have to be of a specific area type. For each region you can define available bandwidths, the delay model, and the re-transmission model, allowing you to mimic the behavior of a "real cell".
However, it seems to me that the level of detail of the transmission model isn't quite up to your desires.

I can further point you to the [doctoral thesis](https://publica.fraunhofer.de/entities/publication/7556e012-c83d-4bf1-9079-c29097cb7def/details) of Robert Protzmann who developed the initial Cell Simulator as part of his thesis.

**Comment by Jean-Tshibangu-jtm** [comment]
Thanks very much,  Dear @schwepmo   for these details.

**Reply by xwh0625** [reply]
Hello，Is it possible to implement 5G features now?

**Reply by hoelger** [reply]
Currently there is the work to integrate ns3's LTE code. You can find the work-in-progress here:

https://github.com/eclipse-mosaic/mosaic/pull/501
https://github.com/mosaic-addons/ns3-federate/pull/11

This will pave the road to implement 5G features in the future :)

**Comment by hoelger** [comment]
The ns3 LTE feature is now available and merged into the main branches.

**Comment by hoelger** [comment]
Thanks to the hard work of @Jacenty00 we now have a first version where omnet is able to support 5G cellular communication!

See branches here:
https://github.com/eclipse-mosaic/mosaic/tree/1132-enable-lte-5g-in-omnet
https://github.com/mosaic-addons/omnetpp-federate/tree/1132-enable-lte-5g-in-omnet
